### PR TITLE
Merge deferred results

### PR DESCRIPTION
**Pull request by sbarker2**
Merging deferred results to publisher and send back to client (part of the defer implementation for non-hydrated queries)

NOTE: This does not apply the transformers yet - this work is in progress, but just note at this point, queries with transformers (for example renames) will not work.

### CODE REVIEW

File: lib/src/main/java/graphql/nadel/engine/NadelExecutionContext.kt [hunk 1]
```diff
@@ -25,10 +25,6 @@ data class NadelExecutionContext internal constructor(
 ) {
     private val serviceContexts = ConcurrentHashMap<String, CompletableFuture<Any?>>()
 
-    @Deprecated("Use incrementalSupport instead", ReplaceWith("incrementalResultSupport"))
-    internal val deferSupport: NadelIncrementalResultSupport
-        get() = incrementalResultSupport
-
```
Comment (sbarker2) [inline lib/src/main/java/graphql/nadel/engine/NadelExecutionContext.kt:31]: Pretty sure I was the only one using this so we can remove the deprecated field now

File: test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/DeferWithLabelTestSnapshot.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                |       slow
+                |     }
+                |   }
+                | }
+                """.trimMargin(),
+                variables = "{}",
+                response = """
+                | {
+                |   "defer": {
+                |     "hello": "helloString"
```
Comment (felipe-gdr) [inline test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/DeferWithLabelTestSnapshot.kt:36]: The initial response should contain the `hasNext` field with value `true`. See example here:
https://github.com/graphql/graphql-over-http/blob/main/rfcs/IncrementalDelivery.md#content-type-multipartmixed

File: test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/DeferWithLabelTestSnapshot.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+     */
+    override val response: ExpectedNadelResponse = ExpectedNadelResponse(
+            response = """
+            | {
+            |   "data": {
+            |     "defer": {
+            |       "hello": "helloString"
+            |     }
+            |   },
+            |   "hasNext": true
```
Comment (felipe-gdr) [inline test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/DeferWithLabelTestSnapshot.kt:80]: Not sure why `hasNext` is present on the final expected response, but not on the service response.

File: test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/MultipleDeferDirectivesTestSnapshot.kt [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                variables = "{}",
+                response = """
+                | {
+                |   "defer": {
+                |     "fastField": "123"
+                |   }
+                | }
+                """.trimMargin(),
+                delayedResponses = listOfJsonStrings(
+                    """
```
[the remark this answers is not in the document]
Comment (felipe-gdr) [reply test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/MultipleDeferDirectivesTestSnapshot.kt:44]: also, how deterministic is this? Not sure that `slowString` will always be the last field to be resolved here.

File: test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/MultipleFieldsinSingleDeferDirectiveTestSnapshot.kt [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                |       "data": {
+                |         "slowField": "slowString",
+                |         "anotherSlowField": 123456789
+                |       }
+                |     }
+                |   ]
+                | }
+                """.trimMargin(),
+            ),
+        )
```
Comment (felipe-gdr) [inline test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/MultipleFieldsinSingleDeferDirectiveTestSnapshot.kt:103]: Some more test cases you could add:
- defer on a field from a type returned as part of a list
- defer on calls to multiple underlying services 
- error cases - what if the initial response results on an error. What if one or more incremental response results in an error?
- extensions - on initial payload and incremental payloads

**Review by felipe-gdr — CHANGES_REQUESTED** [review]
Great stuff - I marked as "requested changes" because of the missing `hasNext` on service call response and so we can add more test scenarios.

File: test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/DeferWithLabelTestSnapshot.kt (hunk 2 above)
Comment (gnawf) [reply test/src/test/kotlin/graphql/nadel/tests/next/fixtures/defer/DeferWithLabelTestSnapshot.kt:36]: Hmm that's probably my fault. This is odd, it doesn't even contain the `data` field.

**Pull request closed without merging**